src/nodejs/lambda-layers/message-util/src/message-util.js

```diff
@@ -11,6 +13,80 @@ const eventGroupId = 'edpub-event-group';
 // const sesAccessKeyId = process.env.SES_ACCESS_KEY_ID;
 // const sesSecretAccessKey = process.env.SES_SECRET_ACCESS_KEY;
 
+// Helper function to read a stream into a buffer
+const streamToBuffer = (stream) => new Promise((resolve, reject) => {
+  const chunks = [];
+  stream.on('data', (chunk) => chunks.push(chunk));
+  stream.on('end', () => resolve(Buffer.concat(chunks)));
+  stream.on('error', reject);
+});
+
+// Helper to fetch attachment as base64
+const getAttachmentAsBase64String = async ({ bucket, key }) => {
+  const s3 = new S3Client();
+
+  try {
+    const response = await s3.send(new GetObjectCommand({ Bucket: bucket, Key: key }));
+    const buffer = await streamToBuffer(response.Body);
+    return buffer.toString('base64');
+  } catch (error) {
+    console.error(`Failed to fetch S3 object: ${error.message}`);
+    throw error;
+  }
+};
+
+// Generate the raw email with attachment
+const getRawFromTemplate = ({
+  subject,
+  from,
+  to,
+  htmlText,
+  plainText,
+  images = [],
+  nasaLogo
+}) => {
+  const nasaLogoPart = nasaLogo
```

Comment:
Maybe easier to work as part of the [EDPUB-1446](https://bugs.earthdata.nasa.gov/browse/EDPUB-1446) but I wonder if it would be better that we generalize the attachment information similar to what's being done for the imagesPart lines below and include the nasa logo in that generalized input. So the template becomes something like
```
--EDPUB_ALTERNATIVE
Content-Type: ${contentType}
Content-ID: <${contentId}>
Content-Transfer-Encoding: base64
Content-Disposition: inline; filename="${name}"

${data}
```